### PR TITLE
Added Happy Number

### PR DESCRIPTION
**Pull request by anonimak**
This is actually my first pull request, I start with an easy problem hehe
I hope i can contribute more
thanks,,

### CODE REVIEW

File: LeetcodeProblems/Algorithms/Happy_Number.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Return true if n is a happy number, and false if not.
+
+Example 1:
+Input: n = 19
+Output: true
+Explanation:
+12 + 92 = 82
+82 + 22 = 68
+62 + 82 = 100
+12 + 02 + 02 = 1
```
Comment (ignacio-chiazzo) [inline LeetcodeProblems/Algorithms/Happy_Number.js:22]: Could you update this using a `^`? The digits together are a bit confusing.

File: LeetcodeProblems/Algorithms/Happy_Number.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * @param {number} n
+ * @return {boolean}
+ */
+var isHappy = function(n) {
+    return checkHappyNumber(n);
+};
+
+function checkHappyNumber(n){
+    strNumber = n.toString();
+    splitNumber = strNumber.split("");
```
Comment (ignacio-chiazzo) [inline LeetcodeProblems/Algorithms/Happy_Number.js:41]: another solution that doesn't require translating the integer to a String would be to 
get the number, divide it by 10, get the digit and repeat the process.

**Review by ignacio-chiazzo — APPROVED** [review]

**Review by ignacio-chiazzo — COMMENTED** [review]
Thank you for contributing!

File: LeetcodeProblems/Algorithms/Happy_Number.js (hunk 1 above)
Comment (anonimak) [reply LeetcodeProblems/Algorithms/Happy_Number.js:22]: sure

File: LeetcodeProblems/Algorithms/Happy_Number.js (hunk 2 above)
Comment (anonimak) [reply LeetcodeProblems/Algorithms/Happy_Number.js:41]: sorry, i don't get it. could you give an example?

Comment (ignacio-chiazzo) [reply LeetcodeProblems/Algorithms/Happy_Number.js:41]:
```
last_digit = n % 10 
n = (n - last_digit) / 10
```

File: LeetcodeProblems/Algorithms/Shuffle_String.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Input: s = "codeleet", indices = [4,5,6,7,0,2,1,3]
+Output: "leetcode"
+Explanation: As shown, "codeleet" becomes "leetcode" after shuffling.
+
+Example 2:
+
+Input: s = "abc", indices = [0,1,2]
+Output: "abc"
+Explanation: After shuffling, each character remains in its position.
+
```
Comment (ignacio-chiazzo) [inline LeetcodeProblems/Algorithms/Shuffle_String.js:19]: I guess this assumes the input indices is a valid input e.g. it cannot be [-1, 1023234234]

**Review by ignacio-chiazzo — APPROVED** [review]
Thank you!

**Pull request merged**